prov/efa/src/efa_hmem.c

```diff
@@ -377,11 +339,14 @@ int efa_domain_hmem_validate_p2p_opt(struct efa_domain *efa_domain, enum fi_hmem
 	 *     PREFERED means a provider should prefer P2P if it is available.
 	 *
 	 * These options does not require that p2p is supported by device,
-	 * nor do they prohibit that p2p is reqruied by implementation. Therefore
-	 * they are always supported.
+	 * nor do they prohibit that p2p is required by implementation. 
+	 * Therefore they are always supported unless p2p is disabled.
 	 */
 	case FI_HMEM_P2P_PREFERRED:
 	case FI_HMEM_P2P_ENABLED:
+		if (OFI_UNLIKELY(ofi_hmem_p2p_disabled()))
+			return -FI_EOPNOTSUPP;
+
```

Comment:
How can we end up here?

Reply:
Uses could set both FI_HMEM_DISABLE_P2P and FI_HMEM_P2P_PREFERRED/FI_HMEM_P2P_ENABLED, which is not supported

Reply:
I think those options are mutually exclusive, isn't that the case?

Reply:
`FI_HMEM_DISABLE_P2P` is also the name of the env var that controls `ofi_hmem_p2p_disabled()`. So the user can have it set (inadvertently or otherwise) while also passing whatever they want for `fi_setopt()`.